crates/compiler/parse/src/ident.rs

```diff
@@ -556,19 +578,20 @@ fn chomp_identifier_chain<'a>(
                 BadIdent::WeirdDotAccess(pos.bump_column(chomped as u32 + width)),
             )),
         }
-    } else if let Ok(('_', _)) = char::from_utf8_slice_start(&buffer[chomped..]) {
-        // we don't allow underscores in the middle of an identifier
-        // but still parse them (and generate a malformed identifier)
-        // to give good error messages for this case
-        Err((
-            chomped as u32 + 1,
-            BadIdent::UnderscoreInMiddle(pos.bump_column(chomped as u32 + 1)),
-        ))
     } else if first_is_uppercase {
         // just one segment, starting with an uppercase letter; that's a tag
         let value = unsafe { std::str::from_utf8_unchecked(&buffer[..chomped]) };
-
-        Ok((chomped as u32, Ident::Tag(value)))
+        if value.contains('_') {
+            // we don't allow underscores in the middle of an identifier
+            // but still parse them (and generate a malformed identifier)
+            // to give good error messages for this case
+            Err((
+                chomped as u32,
+                BadIdent::UnderscoreInMiddle(pos.bump_column(chomped as u32)),
+            ))
```

Comment:
Does this block compilation? If so I would just allow this and then check later on (canonicalization) that there are no consecutive underscores.

Reply:
This is checking for any underscores in uppercase identifiers.  And it does not block compilation. If you look, this is the same way the same error was added for all identifiers before